scipy/optimize/_linprog.py

```diff
@@ -587,7 +591,11 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
     if x0 is not None and meth != "revised simplex":
         warning_message = "x0 is used only when method is 'revised simplex'. "
         warn(warning_message, OptimizeWarning)
-
+        
+    if 'simplex' in meth:
```

Comment:
Unlikely to become a problem, but just in case we want to add another method with "simplex" in the name, check `meth in {'simplex', 'revised simplex'}` instead.